### PR TITLE
Updating Readme

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
check here for [view](https://github.com/GoogleCloudPlatform/gcsfuse/blob/1d285c55b68e7ac32c34ded28d63753e1225a0f1/README.md)

### Testing
N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,193 +1,233 @@
...
+FUSE has a file system interface, it is not like a true POSIX, NFS or CIFS file 
+system on the backend. It is ideal for use cases where Cloud Storage has the right 
+performance and scalability characteristics for an application, and only basic file
+system semantics are missing. When deciding if Cloud Storage FUSE is an appropriate 
+solution, there are some additional differences compared to local file systems that 
+should be taken into account:
+
+**Performance**: Cloud Storage FUSE is a client interface to Cloud Storage and 
+therefore has the same performance characteristics of Cloud Storage. See 
+performance best practices [here](https://docs.google.com/document/d/1s1K5zQBZikeJD5ThAQS13jgjTbMImAdDiTSmfYKJykg/edit#bookmark=id.mk1mwrsv70l1)
```
Comment (raj-prince) [inline README.md:86]: I think, we should use Github performance doc link instead of internal one.

File: README.md [hunk 2]
```diff
@@ -1,193 +1,233 @@
...
-
-If you notice otherwise unreasonable performance, please [file an
-issue][issues].
-
-[issues]: https://github.com/googlecloudplatform/gcsfuse/issues
+Using the [Cloud Storage FUSE CSI driver](https://github.com/GoogleCloudPlatform/gcs-fuse-csi-driver), users get the declarative nature of Kubernetes
+with all infrastructure fully managed by GKE in combination with Cloud Storage. This CSI
+driver relies on Cloud Storage FUSE to mount Cloud storage buckets as file systems on the
+GKE nodes, with the Cloud Storage FUSE deployment and management fully handled by GKE, 
+providing a turn-key experience
```
Comment (raj-prince) [inline README.md:157]: Full stop missing.

File: README.md [hunk 3]
```diff
@@ -1,193 +1,233 @@
...
+[Troubleshooting]:https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/docs/troubleshooting.md
+
+## Limitations and restrictions
+
+Cloud Storage FUSE is not supported for usage with the following workloads or scenarios:
+
+- Workloads that expect POSIX compliance.
+     - Cloud Storage FUSE is not a traditional POSIX filesystem. For a fully supported
+       enterprise grade filesystem, see [Cloud Filestore]. See Key Differences from a 
+       POSIX filesystem and Semantics doc
```
Comment (raj-prince) [inline README.md:197]: Can you please provide the link for key-difference from a traditional file-system.

File: README.md [hunk 4]
```diff
@@ -1,193 +1,233 @@
...
+  
+- Object versioning: Cloud Storage FUSE does not support buckets with object versioning enabled.
+No guarantees are made about the behavior when used with a bucket with this feature enabled, which may
+return non-current object versions.
+- Transcoding: Reading or modifying a file backed by an object with a contentEncoding property set may
+yield surprising results, and no guarantees are made about the behavior. It is recommended that you do not use 
+Cloud Storage FUSE to interact with such objects.
+- Retention policy: Cloud Storage FUSE does not support writing to a bucket with a Retention policy enabled - writes will fail.
+Reading objects from a bucket with a Retention policy enabled is supported, but the bucket should be mounted as Read-Only 
+by passing the -o RO flag during mount.
```
Comment (raj-prince) [inline README.md:213]: Can you put this flag inside `` quote?

**Review by raj-prince — APPROVED** [review]
Please incorporate the comments before submission.

File: README.md (hunk 1 above)
Comment (Tulsishah) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (Tulsishah) [reply README.md]: fixed

File: README.md (hunk 4 above)
Comment (Tulsishah) [reply README.md]: fixed

**Pull request merged**